### PR TITLE
feat(frontend): target api support

### PR DESCRIPTION
**Pull request by artek-koltun**
see https://github.com/opiproject/opi-api/pull/375

Requires https://github.com/opiproject/godpu/pull/327 to pass ci

### CODE REVIEW

File: pkg/kvm/transport.go [hunk 1]
```diff
@@ -38,7 +38,7 @@ func NewNvmeVfiouserTransport(ctrlrDir string) frontend.NvmeTransport {
 }
 
 func (c *nvmeVfiouserTransport) Params(ctrlr *pb.NvmeController, nqn string) (spdk.NvmfSubsystemAddListenerParams, error) {
-	pcieID := ctrlr.Spec.PcieId
```
Comment (artek-koltun) [inline pkg/kvm/transport.go:41]: I am thinking about moving vfiouser transport into frontend in a following PR, so we could test pcie transport with godpu... @glimchb wdyt?

File: pkg/utils/server.go [hunk 2]
```diff
@@ -145,3 +146,12 @@ func spdkMockServerCommunicate(rpc spdk.JSONRPC, l net.Listener, toSend []string
 func ResourceIDToVolumeName(resourceID string) string {
 	return fmt.Sprintf("//storage.opiproject.org/volumes/%s", resourceID)
 }
+
+// OpiAdressFamilyToSpdk converts opi address family to the one used in spdk
+func OpiAdressFamilyToSpdk(adrfam pb.NvmeAddressFamily) string {
+	if adrfam == pb.NvmeAddressFamily_NVME_ADDRESS_FAMILY_UNSPECIFIED {
+		return ""
+	}
```
Comment (glimchb) [inline pkg/utils/server.go:154]: ok for now, maybe better to have explicit switch case for translation between OPI PROTO and SPDK names...

File: README.md [hunk 3]
```diff
@@ -79,7 +79,9 @@ docker run --network=host --rm -it namely/grpc-cli ls   --json_input --json_outp
...
-docker run --network=host --rm -it namely/grpc-cli call --json_input --json_output 10.10.10.10:50051 CreateNvmeController "{parent: '//storage.opiproject.org/subsystems/subsystem2', nvme_controller : {spec : {nvme_controller_id: 2, pcie_id : {physical_function : 0, virtual_function : 0, port_id: 0}, max_nsq:5, max_ncq:5 } }, nvme_controller_id : 'controller1'}"
+docker run --network=host --rm -it namely/grpc-cli call --json_input --json_output 10.10.10.10:50051 CreateNvmeController "{parent: '//storage.opiproject.org/subsystems/subsystem2', nvme_controller : {spec : {nvme_controller_id: 2, pcie_id : {physical_function : 0, virtual_function : 0, port_id: 0}, max_nsq:5, max_ncq:5, 'trtype': 'NVME_TRANSPORT_PCIE' } }, nvme_controller_id : 'controller1'}"
+docker run --network=host --rm -it namely/grpc-cli call --json_input --json_output 10.10.10.10:50051 CreateNvmeSubsystem "{nvme_subsystem : {spec : {nqn: 'nqn.2022-09.io.spdk:opitest3', serial_number: 'myserial2', model_number: 'mymodel2', max_namespaces: 11} }, nvme_subsystem_id : 'subsystem3' }"
```
Comment (glimchb) [inline README.md:83]: please also add GET calls, this verifies they are really created properly

**Review by glimchb — COMMENTED** [review]
few tiny comments inside
next time let's try to make smaller PRs, 20 commits... a lot...
suggestion:
* just add oneof per API change so we can unblock more API changes
* then we can do internal code refactoring

**Review by glimchb — APPROVED** [review]

File: README.md (hunk 3 above)
Comment (artek-koltun) [reply README.md:83]: https://github.com/opiproject/opi-spdk-bridge/pull/690

**Pull request merged**